pkg/odo/cli/component/list.go

```diff
@@ -52,6 +57,65 @@ func (lo *ListOptions) Validate() (err error) {
 
 // Run has the logic to perform the required actions as part of command
 func (lo *ListOptions) Run() (err error) {
+	var ap []component.Component
+	if lo.pathFlag != "" {
+		err := filepath.Walk(lo.pathFlag, func(path string, f os.FileInfo, err error) error {
```

Comment:
Can we put this in some separate function? This `if` statement is getting huge.

Reply:
sure